ui/src/app/settings/components/clusters-list/clusters-list.tsx

```diff
@@ -33,7 +33,7 @@ export const ClustersList = (props: RouteComponentProps<{}>) => {
                                                     <div
                                                         className='argo-table-list__row'
                                                         key={cluster.server}
-                                                        onClick={() => ctx.navigation.goto(`./${encodeURIComponent(cluster.server)}`)}>
+                                                        onClick={() => ctx.navigation.goto(`./${encodeURIComponent(cluster.server)}`, {}, {replace: true})}>
```

Comment:
I think the ClustersList -> Cluster Details page navigation should be recorded in history. Please revert this line change.